homeworks/lesson3/app/src/components/login/login.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Button, FormGroup, FormControl, ControlLabel } from 'react-bootstrap';
+import './login.css';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    
+    //this.props.dispatch(userActions.logout()); // reset login status
+
+    this.state = {
+      email: '',
+      password: '',
+      email_msg: false,
+      password_msg: false,
+      submitted: false
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleMessage = event => {
```

Comment:
That method is to difficult and has many crutches.
So is needed to refactor and think to update all logic of that component.